lib/rules/sort-prop-types.js

```diff
@@ -83,6 +121,45 @@ module.exports = {
         return;
       }
 
+      function fix(fixer) {
+        function sortInSource(allNodes, source) {
+          const originalSource = source;
+          const nodeGroups = allNodes.reduce((acc, curr) => {
+            if (curr.type === 'ExperimentalSpreadProperty' || curr.type === 'SpreadElement') {
+              acc.push([]);
+            } else {
+              acc[acc.length - 1].push(curr);
+            }
+            return acc;
+          }, [[]]);
+
+          nodeGroups.forEach(nodes => {
+            const sortedAttributes = nodes.slice(0).sort(sorter);
+
+            for (let i = nodes.length - 1; i >= 0; i--) {
+              const sortedAttr = sortedAttributes[i];
+              const attr = nodes[i];
+              let sortedAttrText = sourceCode.getText(sortedAttr);
+              if (sortShapeProp && isShapeProp(sortedAttr.value)) {
+                const attrSource = sortInSource(
+                  sortedAttr.value.arguments[0].properties,
```

Comment:
does isShapeProp ensure that the value has a non-empty arguments? What about `PropTypes.shape()`?

Reply:
Good point, I added test cases for this. Also there was already a bug that cause the linting to fail on `PropTypes.shape()`. I included this fix: https://github.com/yannickcr/eslint-plugin-react/pull/1842